### ISSUE TITLE
After updating to v4.22.0 "service not running" message box appears on startup on win10

### ISSUE THREAD

**Issue by BePo65** [docker/for-win#13641]
### Description

After updating to v4.22.0 I keep getting a warning "service is not running" after booting win10.

![ErrorDocker_01](https://github.com/docker/for-win/assets/6582465/a212f537-51d6-476a-966f-df51159e072b)

### Reproduce

1. power on win 10 computer
2. wait a few seconds
3. warning appears in a message box saying:
   "service not running"
   "Docke Desktop service is not running, would you like to start it?Windows will ask you for elevated access."
5. click "start"
6. windows is not asking "for elevated access"
7. docker desktop will run as exxpected

### Expected behavior

before v4.22.0 this message box did not appear.

### docker version

```bash
Client:
 Cloud integration: v1.0.35-desktop+001
 Version:           24.0.5
 API version:       1.43
 Go version:        go1.20.6
 Git commit:        ced0996
 Built:             Fri Jul 21 20:36:24 2023
 OS/Arch:           windows/amd64
 Context:           default

Server: Docker Desktop 4.22.0 (117440)
 Engine:
  Version:          24.0.5
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.6
  Git commit:       a61e2b4
  Built:            Fri Jul 21 20:35:45 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.5
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.20.2-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.6
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scan.exe
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.20.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 5
  Running: 5
  Paused: 0
  Stopped: 0
 Images: 24
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.49-linuxkit-pr
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.841GiB
 Name: docker-desktop
 ID: 99c52d62-9d5d-40f8-ae78-cf457664d91d
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 87
  Goroutines: 100
  System Time: 2023-08-09T08:31:14.2540105Z
  EventsListeners: 11
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

374F0BEE-826D-41F5-B516-1F166C3ADD87/20230809081120

**Comment by JohnWeidmann** [comment]
This is the same issue I am having after yesterday update. Originally app on load was stuck at  "starting engine" - this morning it shows the new updates but..

 Docker Desktop App says engine is running. However trying to manually run containers or /sail up - I immediately get "Docker Engine Not Running" 

I tried to revert back, reinstall docker, update distros, run different default distros etc...Current Default Ubuntu 22, stop and start docker desktop service

OS windows 11 - 64
Diag Id - 66E3DE83-03F9-4299-8D3A-86E4DBB125FE/20230810140504
Docker Desktop 4.22.0 (117440) is currently the newest version available.

Docker Desktop Service - Manaul

Fresh install - no containers running or images installed

Edit: Its building now - not sure if something was updated. I closed > reopened docker. Went to powershell > "docker run hello-world" > went to docker to see if it was installed, hit play button - stopped it > then went back to powershell to run sail up and now its building from docker-compose-yml.

Looks like everything built but now theres an root user error with mysql..

**Comment by MattLichter** [comment]
Seeing this too.  I set up a fresh machine for CI/CD purposes yesterday.  Bare metal install on Windows.  I see this after logging into Windows under our dedicated CI user account, which doesn't have admin privileges.

This issue makes our CI system fragile, as we need to manually attend to the machine after every power outage, windows update (reboot), etc.

I took meticulous notes of the install process, if anyone is having difficulty reproducing.  I'd be happy to share if useful.

In the meantime, can anyone suggest a workaround, where Docker is fully functional after merely logging into Windows?

**Comment by ebriney** [comment]
Hi, we are investigating the issue
@BePo65 Is it happening on every restart? From your diags I see that service is setup in automatic mode so it should start before the user application displaying the popup is started that's weird
@JohnWeidmann did you see the popup above, you're not supposed to as you are in WSL2 mode, in this mode service is not needed
@MattLichter Installer is not aware of user settings, so if they mismatched the `--backend` installer flag it can happen (see https://docs.docker.com/desktop/install/windows-install/#install-from-the-command-line). You can use `--always-run-service` to get previous behaviour

**Comment by BePo65** [comment]
It happens on every reboot of the pc and this on any of my 3 pc.
Restarting DockerDesktop does not show the message box.

I found an older entry on ["SuperUser"](https://superuser.com/questions/1657661/docker-service-is-not-running-windows-will-ask-you-for-elevated-access) that looks similar to my effect. Is it possible that I have a similar effect?
I didn't want to fiddle with access rights, so I did not try th described solution.

**Comment by MattLichter** [comment]
@ebriney Thank you very much!  I reinstalled Docker Desktop for Windows using the following PowerShell command, and my issues were resolved:

`Start-Process -FilePath "Docker Desktop Installer.exe" -ArgumentList "install --quiet --accept-license --backend=windows --always-run-service" -Wait`

**Comment by jason-bivins** [comment]
this issue has been resolved in the 4.23 release - some users experienced the pop-up on the first reboot after installing.
https://docs.docker.com/desktop/release-notes/
